pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -704,6 +725,38 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 	return changed
 }
 
+func delayPublishDomainConfig(ctx *zedmanagerContext, config types.AppInstanceConfig, status *types.AppInstanceStatus, dc *types.DomainConfig) {
+	status.State = types.START_DELAYED
+	uuidStr := status.Key()
+	passedSinceStart := time.Now().Sub(ctx.delayFromTime)
+	timer := time.NewTimer(config.StartDelay - passedSinceStart)
+	status.StartDelay = config.StartDelay
+	if ctx.delayFromTime.Add(config.StartDelay).After(ctx.delayWindowEnds) {
+		ctx.delayWindowEnds = ctx.delayFromTime.Add(config.StartDelay)
+		log.Warnf("@ohm: Added a new VM, extend the delay window (%s)!", ctx.delayWindowEnds.String())
+	}
+	ctx.appsInternalState[uuidStr] = types.AppInstanceInternalStatus{
+		UUID:       uuidStr,
+		DelayTimer: timer,
+	}
+	go func() {
+		log.Warnf("@ohm: Gonna wait now for %s (delay: %s, passed: %s, until: %s)",
+			(status.StartDelay - passedSinceStart).String(), status.StartDelay.String(), passedSinceStart.String(),
+			ctx.delayFromTime.Add(status.StartDelay).String())
+		<-timer.C
+		//nil the timer to prevent it from resetting on ConfigModify
```

Comment:
Somewhere in here it makes sense to check whether the appsInternalState entry for this app has been deleted and if so just return.

Reply:
If the app is removed/deactivated we even do not reach this place, as the timer never fires in this case.

Reply:
What about the case when you start this timer for 10 minutes and 2 minutes later the app instance is deleted?

Reply:
I also stopped the Timer in this case. So we also didn't reach this place.
Anyway, I'm rewriting the whole thing at the moment.